# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом.



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [ ]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [ ]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [ ]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [ ]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [ ]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [ ]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [ ]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [ ]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [ ]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [ ]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [ ]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [ ]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [ ]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [ ]:
data.duplicated().sum()

71

In [ ]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [ ]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [ ]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [ ]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [ ]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [ ]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [ ]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
display(data.head(5)) # посмотрим еще раз на первые 5 строк, чтобы проверить все ли расчетные параметры загрузились

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose total_income_category          purpose_category  
0               покупка жилья                     B  операции с недвижимостью  
1     приобретение автомобиля                     C    операции с автомобилем  
2               покупка жилья                     C  операции с недвижимостью  
3  дополнительное образование                     B     получение образования  
4             сыграть свадьбу                     C        проведение свадьбы

In [ ]:
# для ответа на этот вопрос нам понадобятся 2 столбца: количество детей (children) и имел ли задолженность по возврату кредита (debt)
# столбец family_status вывели для инфо

child_debt = data[['children', 'debt', 'family_status']]
display(child_debt.head(5))

children  debt     family_status
0         1     0   женат / замужем
1         1     0   женат / замужем
2         0     0   женат / замужем
3         3     0   женат / замужем
4         0     0  гражданский брак

In [ ]:
child_pivot = child_debt.pivot_table(index=['children'], values='debt', aggfunc= {'count','sum'})
child_pivot = child_pivot.rename(columns={'count':'Все заемщики','sum':'Должники'})
display(child_pivot)

Все заемщики  Должники
children                        
0                14091      1063
1                 4808       444
2                 2052       194
3                  330        27
4                   41         4
5                    9         0

In [ ]:
# добавим дополнительный расчетный столбец
child_pivot['Доля должников'] = (round(child_pivot['Должники'] / child_pivot['Все заемщики'], 2)) * 100
display(child_pivot)

Все заемщики  Должники  Доля должников
children                                        
0                14091      1063             8.0
1                 4808       444             9.0
2                 2052       194             9.0
3                  330        27             8.0
4                   41         4            10.0
5                    9         0             0.0

**Вывод:**

Отвечая на вопрос: да, по данным таблицы, можем выявить зависимость между количеством детей и возвратом кредита в срок.
Если рассматривать только количественные показатели, то максимальное количество невозвратов(1063) в категории, у которых нет детей. При этом относительный показатель - 8% невозвратов не очень высокий (они более платежеспособные) и идентичен категории заемщиков, у которых 3е детей.

В то же время заемщики, у которых 4 детей хоть и имеют 4 неврозврата из 41, доля просрочки у них самая высокая - 10% и это объясняется сильной финансовой нагрузкой на семью с 4мя детьми.

Семьи с 1 и 2 детьми имеют долю невозвратов 9%, хотя количественно показатели у них невысокие по отношению к общему количеству заемщиков.


#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [ ]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
# для ответа на этот вопрос нам понадобятся 3 столбца: family_status, family_status_id и debt
family_debt = data[['family_status', 'family_status_id', 'debt']]
display(family_debt.head(5))


family_status  family_status_id  debt
0   женат / замужем                 0     0
1   женат / замужем                 0     0
2   женат / замужем                 0     0
3   женат / замужем                 0     0
4  гражданский брак                 1     0

In [ ]:
# построим сводную таблицу
family_pivot = family_debt.pivot_table(index=['family_status'], values='debt', aggfunc= {'count','sum'})
family_pivot = family_pivot.rename(columns={'count':'Все заемщики','sum':'Должники'})
display(family_pivot)


Все заемщики  Должники
family_status                                
Не женат / не замужем          2796       273
в разводе                      1189        84
вдовец / вдова                  951        63
гражданский брак               4134       385
женат / замужем               12261       927

In [ ]:
# добавим дополнительный расчетный столбец
family_pivot['Доля должников'] = (round(family_pivot['Должники'] / family_pivot['Все заемщики'], 2)) * 100
display(family_pivot)

Все заемщики  Должники  Доля должников
family_status                                                
Не женат / не замужем          2796       273            10.0
в разводе                      1189        84             7.0
вдовец / вдова                  951        63             7.0
гражданский брак               4134       385             9.0
женат / замужем               12261       927             8.0

**Вывод:**

Отвечая на вопрос: да мы видим, что присутствует зависимость между возвратом кредита и семейным положением. Минимальная доля должников - 7% в статусе "в разводе" / "вдовец / вдова" , что можно объяснить отстуствием высокой финансовой потребности при проживании одного человека.

Категория заемщиков в статусе "женат / замужем" имеют также низкую долю невозвратов - 8% при высоком абсолютном значении по количеству 927 штук.

Наиболее рисковой категорией являются заемщики в статусе "Не женат / не замужем" и "гражданский брак" 10% и 9% соответственно. Такой показатель можно объяснить тем, что у них снижен уровень отвественности, либо молодой возраст у заемщиков (в силу инфантильности могут забывать платить кредит в срок).

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [ ]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
# для ответа на этот вопрос нам понадобятся 3 столбца: total_income, total_income_category и debt
income_debt = data[['total_income', 'total_income_category', 'debt']]
display(income_debt.head(5))

total_income total_income_category  debt
0        253875                     B     0
1        112080                     C     0
2        145885                     C     0
3        267628                     B     0
4        158616                     C     0

In [ ]:
# построим сводную таблицу
# как помним ранее, то мы производили категоризацию по уровню дохода:

# - 0–30000 — `'E'`;
# - 30001–50000 — `'D'`;
# - 50001–200000 — `'C'`;
# - 200001–1000000 — `'B'`;
# - 1000001 и выше — `'A'`.

income_pivot = income_debt.pivot_table(index=['total_income_category'], values='debt', aggfunc= {'count','sum'})
income_pivot = income_pivot.rename(columns={'count':'Все заемщики','sum':'Должники'})
display(income_pivot)


Все заемщики  Должники
total_income_category                        
A                                25         2
B                              5014       354
C                             15921      1353
D                               349        21
E                                22         2

In [ ]:
# добавим дополнительный расчетный столбец
income_pivot['Доля должников'] = (round(income_pivot['Должники'] / income_pivot['Все заемщики'], 2)) * 100
display(income_pivot)

Все заемщики  Должники  Доля должников
total_income_category                                        
A                                25         2             8.0
B                              5014       354             7.0
C                             15921      1353             8.0
D                               349        21             6.0
E                                22         2             9.0

**Вывод:**

Отвечая на вопрос: судя по сводной таблице выше, нельзя отметить четкую корреляцию между уровнем дохода и возвратом кредита. Данные таблицы больше говорят о том, что существует потребность в кредитовании при уровне дохода от 50 тыс до 200 тыс и поэтому максимальные показатели сосредоточены именно в этом диапазоне в категории "С" (1353 сделки). В то время как клиенты категории "А" и "Е" имеют равные шансы по долгу.

Наиболее рисковой категорией с долей невозврата 8-9% являются заемщики из категории "Е" "С" и "А": первые в силу своего низкого уровня дохода, который может не покрывать платежи по кредиту и общую финансовую нагрузку, вторые и третьи в силу своей высокой финансовой нагрузки (скорее всего у них есть семьи), либо наличие других кредитов.


#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [ ]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
# Ваш код будет здесь. Вы можете создавать новые ячейки.
# для ответа на этот вопрос нам понадобятся 3 столбца: purpose_category, purpose и debt
purpose_debt = data[['purpose_category', 'purpose', 'debt']]
display(purpose_debt.head(5))

purpose_category                     purpose  debt
0  операции с недвижимостью               покупка жилья     0
1    операции с автомобилем     приобретение автомобиля     0
2  операции с недвижимостью               покупка жилья     0
3     получение образования  дополнительное образование     0
4        проведение свадьбы             сыграть свадьбу     0

In [ ]:
# построим сводную таблицу
purpose_pivot = purpose_debt.pivot_table(index=['purpose_category'], values='debt', aggfunc= {'count','sum'})
purpose_pivot = purpose_pivot.rename(columns={'count':'Все заемщики','sum':'Должники'})
display(purpose_pivot)


Все заемщики  Должники
purpose_category                                
операции с автомобилем            4279       400
операции с недвижимостью         10751       780
получение образования             3988       369
проведение свадьбы                2313       183

In [ ]:
# добавим дополнительный расчетный столбец
purpose_pivot['Доля должников'] = (round(purpose_pivot['Должники'] / purpose_pivot['Все заемщики'], 2)) * 100
display(purpose_pivot)

Все заемщики  Должники  Доля должников
purpose_category                                                
операции с автомобилем            4279       400             9.0
операции с недвижимостью         10751       780             7.0
получение образования             3988       369             9.0
проведение свадьбы                2313       183             8.0

**Вывод:**
Отвечая на вопрос: да, мы видим влияние цели кредитования на возврат кредита в срок. Получается, более серьезная цель - "операции с недвижимостью" имеет самый высокий количественный показатель (780 шт), и самый низкий минимальный относительный показатель - 7%. Скорее всего такой низкий показатель обусловлен тем, что это недвижимость и они могут более отвественно погашать задолженность, а иначе может привести к неблагоприятным последствиям и потери места жительства.

Самый рисковой с точки зрения возврата являются цели "операции с автомобилем" и "получение образования" имеют практически идентичные уровни (400 и 369 соотвественно) и доля невозврата 9%. Это объясняется скорее всего, что заемщики небольшой возрастной категории, которые получают образование и у них может быть недостаточно доходов, чтобы вовремя вносить платежи. Аналогично и покупка автомобиля.

Минимальный размер невозвратов 183 сделки имеет цель "проведение свадьбы", но при этом доля составляет 8%, что обусловлено также молодой возрастной категорией заемщиков и невыскоим уровнем дохода.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:*
Наиболее частые причины появления пропусков:
- случайная, то есть при заполнении данных или опроса присутствует "человеческий фактор": забыли поставить значение или пропустили заполнение;
- неслучайная (сокрытие данных), то есть преднамеренное удаление данных из массива данных;
- потерянные, то есть при выгрузке данных могли не учесть какой-то параметр и данные затерлись или потерялись в процессе пересылки между контрагентами, например.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:*
Заполнение пропусков медианным значением более корректно, так как медиана - это середина среди всех значений количественного показателя и отражает объективную картину в данных. В то время, как например, заполнение данных средним значением будет некорректно, особенно в случаях когда есть выбросы по значениям.

### Шаг 4: общий вывод.

Напишите ваш общий вывод.

Кредитным отделом банка были предоставлены данные для анализа по заемщикам - статистика о платежеспособности клиентов. Датасет был представлен 12 показателями и 2 расчетными показателями (по категориям). На входе кредитным отделом была поставлена задача по поиску корреляции между показателми разных категорий и задано 4 вопроса.

В ходе исследования использовались такие методы визуализации и анализа данных, как:
- вывод информации о типах данных
- выявление и удаление пропусков путем расчета медианного значения
- произведена обработка и удаление аномальных значений
- изменение типов данных
- обработка и удаление дубликатов
- произведена категоризация количественных и категориальных показателей
- фильтрация основной таблицы для поиска закономерностей между показателями
- агрегация данных с помощью сводной таблицы
- построение простой гистограммы

По результатам исследования была выявлена закономерность между: количеством детей и возвратом кредита в срок, между семейным положением и возвратом кредита в срок, между уровнем дохода и возвратом кредита в срок и как цели кредита влияют на его возврат в срок.

Также был дан ответ на 2 общих вопроса о данных по наличию пропусков и заполнению их медианным значением.

Выводы для банка: данный набор показателей вполне достаточен для построения скоринговой модели по заемщикам. Из всех представленных строк - 21525 , всего лишь 2174 строки были с отсутствующими значениями по столбцам "общий трудовой стаж" и "ежемесячный доход", что составило 10% от общей выборки. Вцелом это не очень высокий показатель, поэтому было достаточно заполнить данные медианным значением. Количество дубликатов также было несущественно (71 строка или 0,3%).

Из 4х поставленных вопросов по 3 была выявлена очевидная корреляция, а по одному вопросу - закономерность между уровнем дохода и возвратом кредита в срок был получен неоднозначный ответ. Рекомендуется либо дополнить выборку по заемщикам с уровнем дохода из категорий A, D, E, либо убедиться, что данные заполнены корректно и действительно заемщики с уровненм дохода от 50 тыс до 1 млн руб активно пользуются кредитованием.